### PR TITLE
Fix linspace/logspace docstrings: start/end are Number, not float

### LINKED ISSUE #129338: `linspace()` can also use `complex` and `bool` type for `start` and `end` argument against the doc
### 🐛 Describe the bug

[The doc](https://pytorch.org/docs/stable/generated/torch.linspace.html) of `linspace()` says that `float` type can be used for `start` and `end` argument as shown below:

> - start ([float](https://docs.python.org/3/library/functions.html#float) or [Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the starting value for the set of points. If Tensor, it must be 0-dimensional
> - end ([float](https://docs.python.org/3/library/functions.html#float) or [Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor)) – the ending value for the set of points. If Tensor, it must be 0-dimensional

But, `linspace()` can also use `complex` and `bool` for `start` and `end` argument against the doc as shown below:

```python
import torch

torch.linspace(start=0.+0.j, end=5.+0.j, steps=2) # tensor([0.+0.j, 5.+0.j])

torch.linspace(start=False, end=True, steps=2) # tensor([0., 1.])
```

### Versions

```python
import torch

torch.__version__ # 2.3.0+cu121
```

### PR DESCRIPTION
**Pull request by vineethsaivs**
Fixes #129338. Also covers `torch.logspace` (#130254), which has the identical wording.

`torch.linspace` and `torch.logspace` document `start` and `end` as `float or Tensor`:

```
start (float or Tensor): the starting value for the set of points. ...
end (float or Tensor): the ending value for the set of points. ...
```

But this is narrower than what the functions accept and is inconsistent with the same docstring's own dtype note, which already contemplates complex inputs:

```
... and corresponding complex dtype when either is complex.
```

`int` and `bool` work too (`bool` is a subclass of `int`, `int`/`float`/`complex` are all `numbers.Number`). The sibling factory `torch.arange` already documents `start`/`end`/`step` as `Number`.

This aligns the `start`/`end` type label with `Number or Tensor`, matching `arange` and the functions' own complex-aware dtype note. Docstring-only, `+4/-4`, no runtime change. (Scoped to `start`/`end`; `logspace`'s `base` is left unchanged.)

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Comment by vineethsaivs** [comment]
Thanks for the approval @Skylion007! I tried `@pytorchbot merge` as the author but the bot replied that the pull workflow has not been scheduled for the PR (external-author permissions). Since this is a docs-only change to `torch/_torch_docs.py` (+4/-4, no runtime/code impact), would you mind running `@pytorchbot merge` (or `-f`) when you have a moment? Appreciated.

**Comment by vineethsaivs** [comment]
Gentle ping @Skylion007: as an external author I cannot schedule the pull workflow or run the merge bot myself, so this docs-only +4/-4 (EasyCLA green) still needs a maintainer `@pytorchbot merge -f`. Thanks!

**Comment by vineethsaivs** [comment]
Gentle ping @Skylion007 @albanD: this has been approved since 28 Jun and just needs a maintainer to land it. As an external author `@pytorchbot merge` comes back with *"Pull workflow has not been scheduled for the PR yet ... author does not have permissions to run those"*, so I cannot schedule the workflow or run the bot myself.

It is a docs-only +4/-4 change (`linspace`/`logspace` docstrings say `start`/`end` are `float`, but both accept any `Number`, including complex and tensors of 0 dim), EasyCLA is green and it is labelled `topic: not user facing`. `@pytorchbot merge -f "docs-only change"` should be enough.

Happy to close it if the change is not wanted, just let me know.
